### PR TITLE
DAOS-8624 test: Verify io telemetry metrics

### PR DESCRIPTION
**Pull request by mjean308**
Verify the io latency fetch and update metrics

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

File: src/tests/ftest/control/dmg_telemetry_io_latency.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    elif "RP" in oclass:
+        tmp = re.findall(r'\d+', oclass)
+        if tmp:
+            rf = int(tmp[0]) - 1
+    else:
+        rf = 0
+    return rf
+
+
+def convert_to_number(size):
```
Comment (shimizukko) [inline src/tests/ftest/control/dmg_telemetry_io_latency.py:40]: Is this method used in the test?

File: src/tests/ftest/control/dmg_telemetry_io_latency.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        # sum up the latency values
+        for test_metric in test_metrics:
+            metrics[test_metric] = 0
+            for host in self.hostlist_servers:
+                # test assumes one engine per host
+                for rank in self.server_managers[-1].get_host_ranks([host]):
+                    for target in range(self.server_managers[-1].get_config_value("targets")):
+                        value = metrics_data[test_metric][host][str(rank)][str(target)][size]
+                        metrics[test_metric] = metrics[test_metric] + value
+        if ((metrics["engine_io_latency_fetch_max"] >= metrics["engine_io_latency_fetch"]) and
```
Comment (shimizukko) [inline src/tests/ftest/control/dmg_telemetry_io_latency.py:242]: What's the difference between fetch vs. fetch_max/min/mean? Same with latency.

**Review by shimizukko — DISMISSED** [review]
Since this is telemetry test, shouldn't we add it into the telemetry directory?

**Review by shimizukko — DISMISSED** [review]

**Review by dinghwah — APPROVED** [review]
LGTM

**Comment by jolivier23** [comment]
@mjean308 can you please either rerun checkpatch or explain what things it complained about were resolved.   I have to dismiss its review and it seems to have complained about a lot of stuff that you marked as resolved so I'd rather daosbuild left the LGTM comment if it's golden.   I can force land it with just a checkpatch if needed.

File: src/tests/ftest/control/dmg_telemetry_io_latency.py [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+        errors = False
+        self.log.error("Summary of io latency min, max, mean and stddev comparison results:")
+        for item in verification_results:
+            self.log.info("  %s  %s  %s", item[0], item[1], item[2])
+            if item[0] == "FAILED":
+                errors = True
+        if errors:
+            self.fail("Test FAILED")
+
+    @skipForTicket("DAOS-8985")
```
Comment (sylviachanoiyee) [inline src/tests/ftest/control/dmg_telemetry_io_latency.py:342]: DAOS-8985 is resolved. Should this be removed?

Comment (mjean308) [reply src/tests/ftest/control/dmg_telemetry_io_latency.py:342]: yes... i will update that

**Review by shimizukko — APPROVED** [review]

**Review by dinghwah — DISMISSED** [review]
LGTM

**Comment by sylviachanoiyee** [comment]
@mjean308 , I cannot find the telemetry result in build 12 in jenkins. Do you know why the test wasn't run?
Can you alos update DAOS-9031 at the top of the ticket description that a test is skipped and how to re-enable it?

**Review by shimizukko — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]
LGTM

**Pull request merged**